cpp/src/arrow/compute/kernels/filter.cc

```diff
@@ -163,5 +164,91 @@ Status Filter(FunctionContext* ctx, const RecordBatch& batch, const Array& filte
   return Status::OK();
 }
 
+Status Filter(FunctionContext* ctx, const ChunkedArray& values, const Array& filter,
+              std::shared_ptr<ChunkedArray>* out) {
+  if (values.length() != filter.length()) {
+    return Status::Invalid("filter and value array must have identical lengths");
+  }
+  auto num_chunks = values.num_chunks();
+  std::vector<std::shared_ptr<Array>> new_chunks(num_chunks);
+  std::shared_ptr<Array> current_chunk;
+  int64_t offset = 0;
+  int64_t len;
+
+  for (int i = 0; i < num_chunks; i++) {
+    current_chunk = values.chunk(i);
+    len = current_chunk->length();
+    RETURN_NOT_OK(
+        Filter(ctx, *current_chunk, *filter.Slice(offset, len), &new_chunks[i]));
+    offset += len;
+  }
+
+  *out = std::make_shared<ChunkedArray>(std::move(new_chunks));
+  return Status::OK();
```

Comment:
We should create a function that accepts a lambda for chunked evaluation so this logic can be reused in other places. Does not have to happen in this patch